ringpop.thrift

```diff
@@ -0,0 +1,25 @@
+struct Change {
+    1: required string source;
+    2: required string address;
+    3: required string status;
+    4: required double incarnationNumber;
+}
+
+struct JoinResult {
+    1: required list<Change> changes;
+}
+
+struct PingResult {
+    1: required list<Change> changes;
+}
+
+struct PingReqResult {
+    1: required bool pingStatus;
+    2: required list<Change> changes;
+}
+
+service Ringpop {
+    JoinResult join(1:string app, 2:string source, 3:double incarnationNumber);
```

Comment:
if you want, you can add `required` on each input parameter as well.

however I don't recommend `required` because if you decide to deprecate these required fields in the future, you will have to manipulate the existing fields to remove `required` which is anti-pattern in thrift.
thriftify supports only thrift binary protocol for now. `required` doesn't make any difference on the wire. it ensures the field presence during serialization/deserialization. in some other thrift protocols, `required` changes how the field is serialized which makes it even harder to upgrade with requiredness changes.